modules/userId/index.js

```diff
@@ -605,6 +605,57 @@ function getUserIdsAsEids() {
   return createEidsArray(getCombinedSubmoduleIds(initializedSubmodules));
 }
 
+/**
+ * This function will be exposed in global-name-space so that userIds for a source can be exposed
+ * Sample use case is exposing this function to ESP
+ */
+function getEncryptedEidsForSource(source, encrypt, customFunction) {
+  let eids = [];
+  let eidsSignals = {};
+
+  if (isFn(customFunction)) {
+    logInfo(`${MODULE_NAME} - Getting encrypted signal from custom function : ${customFunction.name} & source : ${source} `);
+    // Publishers are expected to define a common function which will be proxy for signal function.
+    let customSignals = customFunction(source);
+    eidsSignals[source] = customSignals ? encryptSignals(customSignals) : null; // by default encrypt using base64 to avoid JSON errors
+  } else {
+    // initialize signal with eids by default
+    eids = (getGlobal()).getUserIdsAsEids();
```

Comment:
@r-schweitzer  Also, USER_IDS_CONFIG is not exposed to userID/index.js. So can we get the USER_IDS_CONFIG by exporting getUserIdConfigs() in eid.js and importing in index.js ?

Reply:
Yeah you would need to do that if you took that route. Again the alternative is to filter the EIDs earlier instead of creating these new functions.

The main thing I want to get away from is running getUserIdsAsEids within getEncryptedEidsForSource like we're currently doing. For each source we're running the same function and getting all the EIDs over and over for each source listed.

Reply:
Approached Used: By defining exposing getUserIdsAsEidsBySource, we are fetching the eids based on source Name passed.